File: ruby/recursive_count.rb
def recursive_count(num = 0)
  # type your code in here
  if num === 10
    return
  elsif num <= 10
    puts num
    recursive_count(num + 1)
  end

  # ALTERNATE SOLUTION
  # return if num >= 10
  # puts num
  # recursive_count(num + 1)
end

if __FILE__ == $PROGRAM_NAME
  recursive_count
end

# OPTIONAL
# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/recursive_count.js
function recursiveCount(num = 0) {
  // type your code here
  // let count = 0;

  // while(count < 10) {
  //   console.log(count);
  //   ++count;
  // }
  if(num >= 10) {
    // console.log(num);
  } else if(num < 0) {
    console.log('less than 0');
  } else {
    console.log(num);
    recursiveCount(num + 1);
    // recursiveCount(++num);
    // ++num increments then invokes the function/loop
    // num++ invokes the function/loop then increments
  }
}

if (require.main === module) {
  recursiveCount();
  // recursiveCount(10);
}

module.exports = recursiveCount;

// OPTIONAL
// Please add your pseudocode to this file
// And a written explanation of your solution
